src/main/java/tutorlink/commandpackage/DeleteStudentCommand.java

```diff
@@ -0,0 +1,31 @@
+package tutorlink.commandpackage;
+
+import tutorlink.listpackage.StudentList;
+import tutorlink.resultpackage.CommandResult;
+
+public class DeleteStudentCommand extends FindStudentCommand{
+
+    public static final String COMMAND_WORD = "delete_student";
+    public static final String REGEX = "^delete_student (i/(A\\d{7}[A-Z])?( n/[\\w\\d]+)?|n/[\\w\\d]" +
+            "+( i/(A\\d{7}[A-Z])?)?)( )?$";
+
+    private static final String SUCCESS_MESSAGE = "Student %s successfully deleted";
+
+    public DeleteStudentCommand(String name, String matricNumber) {
```

Comment:
It is good that you incorporated name into the DeleteStudentCommand. Have you considered allowing the deletion using only either name or matricNumber? 

You can try overloading the constructor with signatures containing only one of both.

Reply:
Hi thanks for your review. That is indeed possible, although it would be risky to delete only using name as a key since there could be duplicate names in the StudentList.